### DISCUSSION TITLE
Text size

### DISCUSSION THREAD

**Discussion by valleymouth** [vsg-dev/VulkanSceneGraph#619] [Q&A] [answered]
Hi guys,
Very simple question: how do I control the size of a text? Should I just use a transform to scale it up/down?

**Comment by robertosfield** [comment, marked as answer]
The [vsg::StandardLayout](https://github.com/vsg-dev/VulkanSceneGraph/blob/master/include/vsg/text/StandardLayout.h#L21) class is the normal to set the position, orientation and size of the text. The orientation and size are setting is combined into the StandardLayout.horizontal and StandardLayout.vertical members that defines the two axis and scale of each glyph.  The length of the horizontal and vertical effectively set the size.

The [vsgtext](https://github.com/vsg-dev/vsgExamples/blob/master/examples/text/vsgtext/vsgtext.cpp#L267) example illustrates setup using StandardLayout:

```
            auto layout = vsg::StandardLayout::create();
            layout->horizontalAlignment = vsg::StandardLayout::CENTER_ALIGNMENT;
            layout->position = vsg::vec3(6.0, 0.0, 0.0);
            layout->horizontal = vsg::vec3(1.0, 0.0, 0.0); // horizontal size of single glyph is 1.0
            layout->vertical = vsg::vec3(0.0, 1.0, 0.0); // vertical size of single glyph is 1.0
            layout->color = vsg::vec4(1.0, 1.0, 1.0, 1.0);
            layout->outlineWidth = 0.1;
            layout->billboard = true;

            auto text = vsg::Text::create();
            text->text = vsg::stringValue::create("VulkanSceneGraph now\nhas SDF text support.");
            text->font = font;
            text->layout = layout;
            text->setup(0, options);
```

You can also decorate a vsg::Text/TextGroup with a MatrixTransform to scale them, if you want to vary the size dynamically then that's the way I'd do it.

**Reply by valleymouth** [reply]
Thanks!
